### DISCUSSION TITLE
Can use Sqlsync in executable file instead of web browser

### DISCUSSION THREAD

**Discussion by mrsavari** [orbitinghail/sqlsync#59] [Q&A]
**Can use Sqlsync in executable file instead of web browser .**
For offline first implementation?

**Comment by carlsverre** [comment]
Currently it's possible but not super easy. You can see all the code you would require to do this here: https://github.com/orbitinghail/sqlsync/blob/main/lib/sqlsync/examples/end-to-end-local-net.rs

But it's fairly manual at the moment.

What's your use case?

**Reply by mrsavari** [reply]
I have a rust online multi tenant service witch some of my customers demanding offline solution so I did ported to offline mode using tauri but the syncing database part is missing for now.

**Reply by mrsavari** [reply]
@carlsverre is it possible to remove wasm with native implementation because my use case does not involve web browsers and if i need to keep wasm does it make performance impacts on my application?

**Reply by carlsverre** [reply]
Gotcha, that sounds like a great use case. How large is the data of each tenant? Feel free to contact me (carl at orbitinghail.dev) if you want to jump on a call.

As for removing wasm, it's possible but not trivial. There is some code that would need to be changed, but also you need to ensure that mutations can run everywhere. On the plus side the mutations don't have to be precisely the same which makes implementation a bit easier. Probably best to discuss over a call.
